### DISCUSSION TITLE
Obsolete Mode.SetMaterial()

### DISCUSSION THREAD

**Discussion by llfab** [StereoKit/StereoKit#322] [Q&A]
I am loading a Model from an STL file and draw that model with a custom material using transparency. This is done with Model.SetMaterial(0, myMaterial); which is marked as obsolete:

'Model.SetMaterial(int, Material)' is obsolete: 'For removal in v0.4. Use Nodes/Visuals.Material instead.'

I was not able to find details how that shall be changed to Nodes/Visuals.Material. Could you guide me to that?

**Comment by maluoi** [comment]
You should be able to do `model.Visuals[0].material = yourMaterial`! In the case of an STL file, `model.RootNode` and `model.Nodes[0]` should also be equivalent, but `Visuals[0]` is probably the most appropriate in this case :)

If you check the docs for `Model`, there's a number of examples related to working with the newer Node based APIs.
